vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -99,17 +99,43 @@
      */
     // <editor-fold desc="Configuration">
 
+    private static class LazyInitializationProvider {
+
+        @NotNull
+        private final Configuration mConfiguration;
+
+        public LazyInitializationProvider(@NotNull Configuration configuration) {
+            mConfiguration = configuration;
+        }
+
+        @NotNull
+        public Configuration getConfiguration() {
+            return mConfiguration;
+        }
+    }
+
+    private static LazyInitializationProvider sLazyInitializationProvider;
     private static VimeoClient mSharedInstance;
 
     public static VimeoClient getInstance() {
         if (mSharedInstance == null) {
+            if (sLazyInitializationProvider != null) {
+                initialize(sLazyInitializationProvider.getConfiguration());
+                if (mSharedInstance != null) {
+                    return mSharedInstance;
+                }
+            }
             throw new AssertionError("Instance must be configured before use");
         }
 
         return mSharedInstance;
     }
 
-    public static void initialize(Configuration configuration) {
+    public static void initializeLazily(@NotNull Configuration configuration) {
```

Comment:
Do we need to expose this to the consumer of this library? If we just replace the `initialize` method with this implementation, then lazy loading will work with existing code (the get instance code will need to create the new instance instead of calling initialize)